src/components/v2/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { Suspense, useEffect, useState } from 'react';
+import { useRouteMatch, useParams, Redirect } from 'react-router';
+import { Switch, Route } from 'react-router-dom';
+import { URLS } from '../../config';
+import { DetailsProgressing } from '../common';
+import './lib/bootstrap-grid.min.css';
+import ValuesComponent from './values/ChartValues.component';
+import AppHeaderComponent from './headers/AppHeader.component';
+import ChartHeaderComponent from './headers/ChartHeader.component';
+import { getInstalledAppDetail, getInstalledChartDetail } from './appDetails/appDetails.api';
+import AppDetailsComponent from './appDetails/AppDetails.component';
+import { EnvType } from './appDetails/appDetails.type';
+import IndexStore from './appDetails/index.store';
+import ErrorImage from './assets/icons/ic-404-error.png';
+
+let initTimer = null;
+
+function RouterComponent({ envType }) {
+    const [isLoading, setIsLoading] = useState(true);
+    const [isError, setIsError] = useState(false);
+    const params = useParams<{ appId: string; envId: string; nodeType: string }>();
+    const { path } = useRouteMatch();
+
+    useEffect(() => {
+        IndexStore.setEnvDetails(envType, +params.appId, +params.envId);
+
+        setIsLoading(true);
+
+        if (initTimer) {
+            clearTimeout(initTimer);
+        }
+
+        init();
+    }, [params.appId, params.envId]);
+
+    // clearing the timer on component unmount
+    useEffect(() => {
+        return (): void => {
+            if (initTimer) {
+                clearTimeout(initTimer);
+            }
+        };
+    }, []);
+
+    const init = async () => {
```

Comment:
handle tab update also.. as some of the open tab object might have gotten deleted